x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/select_system_prompt/index.tsx

```diff
@@ -122,10 +129,30 @@ const SelectSystemPromptComponent: React.FC<Props> = ({
       if (onSystemPromptSelectionChange != null) {
         onSystemPromptSelectionChange(selectedSystemPromptId);
       }
-      setSelectedSystemPrompt(selectedSystemPromptId);
+      const result = await setSelectedSystemPrompt(selectedSystemPromptId);
+      if (result) {
+        setConversationSettings?.((prev: Record<string, Conversation>) => {
+          const newConversationsSettings = Object.entries(prev).reduce<
+            Record<string, Conversation>
+          >((acc, [key, convo]) => {
+            if (result.title === convo.title) {
+              acc[result.id] = result;
+            } else {
+              acc[key] = convo;
+            }
+            return acc;
+          }, {});
+          return newConversationsSettings;
+        });
+        onSelectedConversationChange?.(result);
+        setConversationsSettingsBulkActions?.({});
+      }
```

Comment:
We have to sync the newly created conversation back to conversationSettings, selectedConversation and bulkActions. Otherwise the selected conversation wouldn't get the latest state and has conflict:

<img width="2538" alt="Screenshot 2024-07-15 at 19 31 27" src="https://github.com/user-attachments/assets/d9fffa36-9704-4808-8d84-c57bce6d85ca">
<img width="2559" alt="Screenshot 2024-07-15 at 19 31 03" src="https://github.com/user-attachments/assets/8f8a5551-c17b-4928-8cfa-2cca47ed01b6">


After:

https://github.com/user-attachments/assets/6a0f9286-5c2e-48ec-99ed-f1d3f20aef1f


https://github.com/user-attachments/assets/5634ddaf-6e3f-470b-8d20-e3fd4f84be78